libafl/src/inputs/encoded.rs

```diff
@@ -93,6 +142,74 @@ impl TokenInputEncoderDecoder {
+                self.id_table.insert(self.next_id, " ".to_string());
+                self.next_id += 1;
+                self.token_table.insert("\t".to_string(), self.next_id);
+                self.id_table.insert(self.next_id, "\t".to_string());
+                self.next_id += 1;
+                self.token_table.insert("\n".to_string(), self.next_id);
+                self.id_table.insert(self.next_id, "\n".to_string());
+                self.next_id += 1;
+                self.token_table.insert("\r\n".to_string(), self.next_id);
+                self.id_table.insert(self.next_id, "\r\n".to_string());
+                self.next_id += 1;
+                self.token_table.insert("  ".to_string(), self.next_id);
+                self.id_table.insert(self.next_id, "  ".to_string());
+                self.next_id += 1;
+                self.token_table.insert("\t\t".to_string(), self.next_id);
+                self.id_table.insert(self.next_id, "\t\t".to_string());
+                self.next_id += 1;
+                self.token_table.insert("\n\n".to_string(), self.next_id);
+                self.id_table.insert(self.next_id, "\n\n".to_string());
+                self.next_id += 1;
+                self.token_table
+                    .insert("\r\n\r\n".to_string(), self.next_id);
+                self.id_table.insert(self.next_id, "\r\n\r\n".to_string());
+                self.next_id += 1;
+                self.token_table.insert("    ".to_string(), self.next_id);
+                self.id_table.insert(self.next_id, "    ".to_string());
+                self.next_id += 1;
+                self.token_table
+                    .insert("\t\t\t\t".to_string(), self.next_id);
+                self.id_table.insert(self.next_id, "\t\t\t\t".to_string());
+                self.next_id += 1;
+                self.token_table
+                    .insert("\n\n\n\n".to_string(), self.next_id);
+                self.id_table.insert(self.next_id, "\n\n\n\n".to_string());
+                self.next_id += 1;
+                self.token_table
+                    .insert("\r\n\r\n".to_string(), self.next_id);
+                self.id_table
+                    .insert(self.next_id, "\r\n\r\n\r\n\r\n".to_string());
+                self.next_id += 1;
+                self.max_whitespace_id = self.next_id;
+
+                // To be able to also insert single quote types during mutation
+                // we also add these
+                self.token_table.insert("\"".to_string(), self.next_id);
+                self.id_table.insert(self.next_id, "\"".to_string());
+                self.next_id += 1;
+                self.token_table.insert("'".to_string(), self.next_id);
+                self.id_table.insert(self.next_id, "'".to_string());
+                self.next_id += 1;
+            }
+            self.encoding_type = enc_type;
+        } else {
+            // TODO: this needs an else that errors
```

Comment:
For error, you want to change the function to return a `Result<(), Error>` and then simply return an error here, like Error::illegal_state(..) or whatever fits